botorch/acquisition/knowledge_gradient.py

```diff
@@ -428,16 +443,44 @@ def forward(self, X: Tensor) -> Tensor:
         return values.mean(dim=0)
 
 
+class ProjectedValueFunction(AcquisitionFunction):
+    r"""
+    Defines a wrapper around the value function that incorporates the project
+    operator.
+    """
+
+    def __init__(
+        self,
+        base_value_function: AcquisitionFunction,
+        project: Callable[[Tensor], Tensor],
+    ):
+        super().__init__(base_value_function.model)
+        self.base_value_function = base_value_function
+        self.project = project
+        self.objective = base_value_function.objective
+        self.sampler = getattr(base_value_function, "sampler", None)
+
+    def forward(self, X: Tensor) -> Tensor:
+        return self.base_value_function(self.project(X))
+
+
 def _get_value_function(
     model: Model,
     objective: Optional[Union[MCAcquisitionObjective, ScalarizedObjective]] = None,
     sampler: Optional[MCSampler] = None,
+    project: Optional[Callable[[Tensor], Tensor]] = None,
 ) -> AcquisitionFunction:
     r"""Construct value function (i.e. inner acquisition function)."""
     if isinstance(objective, MCAcquisitionObjective):
-        return qSimpleRegret(model=model, sampler=sampler, objective=objective)
+        base_value_function = qSimpleRegret(
+            model=model, sampler=sampler, objective=objective
+        )
+    else:
+        base_value_function = PosteriorMean(model=model, objective=objective)
+    if project is None:
+        return base_value_function
     else:
-        return PosteriorMean(model=model, objective=objective)
+        return ProjectedValueFunction(base_value_function, project)
```

Comment:
Our convention is to call with explicit kwargs everywhere (where it makes sense...)

```suggestion
        return ProjectedValueFunction(
            base_value_function=base_value_function,
            project=project,
        )
```